### PR TITLE
feat!: use vue-component-meta

### PR DESCRIPTION
**Pull request by stafyniaksacha**
⚠️ This introduces breaking change in schemas ([more details here]( https://github.com/johnsoncodehk/volar/blob/master/packages/vue-component-meta/tests/index.spec.ts#L30))
 
This is the first implementation of [vue-component-meta](https://github.com/johnsoncodehk/volar/tree/master/packages/vue-component-meta) to extract `props`/`events`/`slots`/`exposed` using typescript vue language service parser from `volar`
_- replaces #23_

It can force usage of typescript in javascript components (thanks that vue3 is written in typescript) so detection work, but it needs a `tsconfig.json` in the root of the project. (can be achieved within vue-component-meta)

`#nuxt-component-meta` virtual module is exposed, so we can use meta inside the project without using a `$fetch` (useful to create components documentation)
_- replaces #32_

- [x] tests (awaiting feedback before updating them)

### CODE REVIEW

File: playground/nuxt.config.ts [hunk 1]
```diff
@@ -13,5 +13,11 @@ export default defineNuxtConfig({
   },
   modules: [
     nuxtMetaModule
-  ]
+  ],
+  componentMeta: {
+    checkerOptions: {
+      forceUseTs: true,
+      schema: {}
```
Comment (stafyniaksacha) [inline playground/nuxt.config.ts:20]: `schema` is required in order to extract the schema, maybe we should use `defu` to have default options?

Comment (farnabaz) [reply playground/nuxt.config.ts:20]: We can provide default value using `defaults` key


```ts
export default defineNuxtModule<ModuleOptions>({
  meta: {
    name: 'nuxt-component-meta',
    configKey: 'componentMeta'
  },
  defaults: {
    checkerOptions: {}
  },
  setup() { ...}
})
```

**Comment by farnabaz** [comment]
Generally LGTM 👍 , but there are some issues we should deal with:

- Default values of props are missing
- In normal components(`TestComponent`), all props are marked as required

Also, I would love to have @kevinmarrec review, about the schema and possible breaking changes for the editor.

**Comment by stafyniaksacha** [comment]
> * Default values of props are missing
> * In normal components(`TestComponent`), all props are marked as required

Should be resolved with https://github.com/johnsoncodehk/volar/pull/1665

**Comment by Tahul** [comment]
@kevinmarrec @farnabaz ; could we state on this PR quickly?

I would love to see this happen to iterate on Studio Elements components documentation we're building.

**Comment by farnabaz** [comment]
Everything looks good to me, The only remaining thing is the tests.
@stafyniaksacha Do you mind updating tests?

**Comment by stafyniaksacha** [comment]
> Everything looks good to me, The only remaining thing is the tests. @stafyniaksacha Do you mind updating tests?

Yep, I'm on it.

I've found that the library is not able to load components from `node_modules` (this affects autoloaded components library)
I don't know if this is a required feature for now, but I think this will be fixed upstream on `vue-component-meta`

**Comment by Tahul** [comment]
Hey @stafyniaksacha ; noticed some changes 😄

Is this PR ready for review again?

I'm willing to have this ASAP for usage in multiple Nuxt packages, that work is amazing!

**Comment by stafyniaksacha** [comment]
Hey @Tahul ! Yes pretty much ready, waiting on next release of vue-component-meta, so we can get rid of `tsconfig.json` that is required now.

This is why the tests are failing, we can not create one in `tests/fixtures/base` because `@nuxt/test-utils` will create a random folder under `.nuxt`

```json
{
  "extends": "./.nuxt/random-xxx/tsconfig.json"
}
```

https://github.com/johnsoncodehk/volar/blob/master/packages/vue-component-meta/tests/index.spec.ts#L684-L693

**Comment by stafyniaksacha** [comment]
Hey @Tahul 
That's should be ready ;)

**Review by farnabaz — APPROVED** [review]
Thanks ❤️

**Pull request merged**